### PR TITLE
Adding filters to elide-js

### PR DESCRIPTION
**Pull request by aruntomsy**

### CODE REVIEW

**Comment by clayreimann** [comment]
You missed `memorystore.js`

**Comment by yahoocla** [comment]
CLA is valid!

File: lib/elide.js [hunk 1]
```diff
@@ -241,6 +241,14 @@ class Elide {
    */
   find(model, id, opts) {
     opts = opts || {};
+    var sparseFields = opts['fields'];
+    var filterFields = opts['filters'];
+    let store = this._getStoreForModel(model);
+    return new Query(store, model, id, opts);
+  }
+
+  find(model, id, opts) {
```
Comment (clayreimann) [inline lib/elide.js:243]: what is this function?

File: spec/stores/jsonapidatastore.spec.js [hunk 2]
```diff
@@ -287,6 +290,36 @@ describe('JsonApiDatastore', function() {
     });
   });
 
+  it('should only fetch literary fiction books', function() {
+    var store = new JsonApiDatastore(ES6Promise, undefined, booksAndAuthorsURL, booksAndAuthorsModels);
+
+    var q = new Query(store, 'book', {filters: {book: [ {attribute: 'genre'}, {operator: "IN"}, {value: "Literary Fiction"} ]}});
```
Comment (clayreimann) [inline spec/stores/jsonapidatastore.spec.js]: This is incorrect `new Query(store, 'book', undefined, {...})`

File: spec/stores/jsonapidatastore.spec.js [hunk 3]
```diff
@@ -287,6 +290,36 @@ describe('JsonApiDatastore', function() {
     });
   });
 
+  it('should only fetch literary fiction books', function() {
```
Comment (clayreimann) [inline spec/stores/jsonapidatastore.spec.js]: need to include `done` callback

File: spec/stores/jsonapidatastore.spec.js [hunk 4]
```diff
@@ -287,6 +290,36 @@ describe('JsonApiDatastore', function() {
...
+    var store = new JsonApiDatastore(ES6Promise, undefined, booksAndAuthorsURL, booksAndAuthorsModels);
+
+    var q = new Query(store, 'book', {filters: {book: [ {attribute: 'genre'}, {operator: "IN"}, {value: "Literary Fiction"} ]}});
+    store.find(q).then(function(result) {
+      return expect(store.find(q))
+        .to.eventually.contain.deep.members([book1, book3, cat1]);
+    });
```
Comment (clayreimann) [inline spec/stores/jsonapidatastore.spec.js:310]: need to `catch(done)`

**Comment by clayreimann** [comment]
:+1:

File: README.md [hunk 5]
```diff
@@ -97,11 +97,13 @@ var options = {
 
 var elide = new Elide(schema, options);
 ```
-#### find(`model`, `id`) → Promise(`result`)
+#### find(`model`, `id`, `opts`) → Promise(`result`)
 `model` - the name of the model (or property for nested queries) to search
 
 `id` - (optional) the id to find (leave blank for collections)
 
+`opts` - (optional) options for fields and filters
```
Comment (clayreimann) [inline README.md]: Can you specify the format please.

**Comment by clayreimann** [comment]
:+1:

**Pull request merged**